src/main/java/bio/terra/service/dataset/flight/ingest/IngestUtils.java

```diff
@@ -137,40 +136,36 @@ private static void validateHost(String host, String URL) {
     if (!actualHostURL.equals(expectedHostURL)) {
       throw new InvalidBlobURLException(
           "Ingest source is not a valid blob url: '"
-              + URL
+              + url
               + "'. "
               + "The host is expected to take the following format: {storageAccountName}.blob.core.windows.net");
     }
   }
 
   private static void validateContainerAndBlobNames(
-      String containerName, String blobName, String URL) {
+      String containerName, String blobName, String url) {
+
+    String azureContainerRegex = "^[a-z0-9](?!.*--)[a-z0-9-]{1,61}[a-z0-9]$";
+
+    if (!containerName.matches(azureContainerRegex)) {
+      throw new InvalidBlobURLException(
+          "Ingest source is not a valid blob url: '"
+              + url
+              + "'. "
+              + " The container and each blob name must meet the following requirements: "
```

Comment:
I would update this exception to indicate an issue only with the container name, not the blob names.

Reply:
great call